.github/workflows/ci.yml

```diff
@@ -1,8 +1,10 @@
 name: CI lints and tests
-on:
-  push:
-    branches:
-      - "*"
+name: CI lints and tests
+  on:
```

Comment:
looks like wrong indentation caused workflow to fail.

Reply:
Good catch! 😅

Reply:
ci is fixed on main now. you can rebase :)

Reply:
Okay things have been reworked here; added a flag to pass down so UDFs can also be used with proper permission checking.

Reply:
Can't we move `ensure_permission` check to hook.rs and udfs? That way dest receiver does not hit into permission checks during copy and regular copy to/from is checked properly.